datasource/threedi_results.py

```diff
@@ -93,10 +95,30 @@ def available_aggregation_vars(self):
         available_aggregation_vars = available_vars & whitelist_vars
         return list(available_aggregation_vars)
 
+    @cached_property
+    def available_water_quality_vars(self):
+        """Return a list of available variables from 'water_quality_results_3di.nc'."""
+        ga = self.water_quality_result_admin
+        if not ga:
+            return []
+        # concentration variable in water quality results
+        concentration_parameter = "concentration"
+        model_instance = ga.get_model_instance_by_field_name(concentration_parameter)
+        if not model_instance:
+            return []
+        substance_name = model_instance.name
+        substance_unit = model_instance.units
+        substance = {
+            "name": substance_name,
+            "unit": substance_unit,
+            "parameters": concentration_parameter,
+        }
+        return [substance]
```

Comment:
So the other "available..." cached properties return a list of str, but here a list of dict is returned. I'm wondering, is there a compelling reason for this, because this discrepancy will make many things more complicated building from here.

Reply:
You are right. They are stored differently. The other `available_..._vars` are stored as string and then later their information is retrieved from `result_constants.py`. These information is hardcoded for different types of variables. However, for the case of substance concentration, the name and unit of the substance are determined by the user. For example, a substance can be either `Benzene hexachloride` or `Dissolved organic carbon` or sth else. 

I can take a further look to see if I can move the logic here elsewhere, for example to the `generate_parameter_config` util function. Then we keep the `available...vars` consistently. What do you think?